std/assembly/util/casemap.ts

```diff
@@ -0,0 +1,496 @@
+  0x1F94,  0x1F2C, 0x0399, 0x0000,
+  0x1F95,  0x1F2D, 0x0399, 0x0000,
+  0x1F96,  0x1F2E, 0x0399, 0x0000,
+  0x1F97,  0x1F2F, 0x0399, 0x0000,
+  0x1F98,  0x1F28, 0x0399, 0x0000,
+  0x1F99,  0x1F29, 0x0399, 0x0000,
+  0x1F9A,  0x1F2A, 0x0399, 0x0000,
+  0x1F9B,  0x1F2B, 0x0399, 0x0000,
+  0x1F9C,  0x1F2C, 0x0399, 0x0000,
+  0x1F9D,  0x1F2D, 0x0399, 0x0000,
+  0x1F9E,  0x1F2E, 0x0399, 0x0000,
+  0x1F9F,  0x1F2F, 0x0399, 0x0000,
+  0x1FA0,  0x1F68, 0x0399, 0x0000,
+  0x1FA1,  0x1F69, 0x0399, 0x0000,
+  0x1FA2,  0x1F6A, 0x0399, 0x0000,
+  0x1FA3,  0x1F6B, 0x0399, 0x0000,
+  0x1FA4,  0x1F6C, 0x0399, 0x0000,
+  0x1FA5,  0x1F6D, 0x0399, 0x0000,
+  0x1FA6,  0x1F6E, 0x0399, 0x0000,
+  0x1FA7,  0x1F6F, 0x0399, 0x0000,
+  0x1FA8,  0x1F68, 0x0399, 0x0000,
+  0x1FA9,  0x1F69, 0x0399, 0x0000,
+  0x1FAA,  0x1F6A, 0x0399, 0x0000,
+  0x1FAB,  0x1F6B, 0x0399, 0x0000,
+  0x1FAC,  0x1F6C, 0x0399, 0x0000,
+  0x1FAD,  0x1F6D, 0x0399, 0x0000,
+  0x1FAE,  0x1F6E, 0x0399, 0x0000,
+  0x1FAF,  0x1F6F, 0x0399, 0x0000,
+  0x1FB2,  0x1FBA, 0x0399, 0x0000,
+  0x1FB3,  0x0391, 0x0399, 0x0000,
+  0x1FB4,  0x0386, 0x0399, 0x0000,
+  0x1FB6,  0x0391, 0x0342, 0x0000,
+  0x1FB7,  0x0391, 0x0342, 0x0399,
+  0x1FBC,  0x0391, 0x0399, 0x0000,
+  0x1FC2,  0x1FCA, 0x0399, 0x0000,
+  0x1FC3,  0x0397, 0x0399, 0x0000,
+  0x1FC4,  0x0389, 0x0399, 0x0000,
+  0x1FC6,  0x0397, 0x0342, 0x0000,
+  0x1FC7,  0x0397, 0x0342, 0x0399,
+  0x1FCC,  0x0397, 0x0399, 0x0000,
+  0x1FD2,  0x0399, 0x0308, 0x0300,
+  0x1FD3,  0x0399, 0x0308, 0x0301,
+  0x1FD6,  0x0399, 0x0342, 0x0000,
+  0x1FD7,  0x0399, 0x0308, 0x0342,
+  0x1FE2,  0x03A5, 0x0308, 0x0300,
+  0x1FE3,  0x03A5, 0x0308, 0x0301,
+  0x1FE4,  0x03A1, 0x0313, 0x0000,
+  0x1FE6,  0x03A5, 0x0342, 0x0000,
+  0x1FE7,  0x03A5, 0x0308, 0x0342,
+  0x1FF2,  0x1FFA, 0x0399, 0x0000,
+  0x1FF3,  0x03A9, 0x0399, 0x0000,
+  0x1FF4,  0x038F, 0x0399, 0x0000,
+  0x1FF6,  0x03A9, 0x0342, 0x0000,
+  0x1FF7,  0x03A9, 0x0342, 0x0399,
+  0x1FFC,  0x03A9, 0x0399, 0x0000,
+  0xFB00,  0x0046, 0x0046, 0x0000,
+  0xFB01,  0x0046, 0x0049, 0x0000,
+  0xFB02,  0x0046, 0x004C, 0x0000,
+  0xFB03,  0x0046, 0x0046, 0x0049,
+  0xFB04,  0x0046, 0x0046, 0x004C,
+  0xFB05,  0x0053, 0x0054, 0x0000,
+  0xFB06,  0x0053, 0x0054, 0x0000,
+  0xFB13,  0x0544, 0x0546, 0x0000,
+  0xFB14,  0x0544, 0x0535, 0x0000,
+  0xFB15,  0x0544, 0x053B, 0x0000,
+  0xFB16,  0x054E, 0x0546, 0x0000,
+  0xFB17,  0x0544, 0x053D, 0x0000
+];
+
+// @ts-ignore: decorator
+@lazy const mt: i32[] = [2048, 342, 57];
+
+// Special binary search routine for Special Casing Tables
+// @ts-ignore: decorator
+@inline
+export function bsearch(key: u32, ptr: usize, max: i32): i32 {
+  var min = 0;
+  while (min <= max) {
+    let mid = (min + max) >>> 3 << 2;
+    let cmp = load<u16>(ptr + (mid << alignof<u16>())) - key;
+    if (cmp == 0) return mid; // found
+    else if (cmp >>> 31) min = mid + 4; // < 0
+    else max = mid - 4; // > 0
+  }
+  return -1; // not found
+}
+
+// See: musl/tree/src/ctype/towctrans.c
```

Comment:
Did look at the musl file, but don't immediately see any similarities. Is this correct?

Reply:
Yeah, fixed